### PR TITLE
Fix weak vtable/type_info for EndpointInfo and ConnectionInfo

### LINKED ISSUE #2071: type_info error on macos
When debugging Ice/info on macos, I get the following message from lldb:

```
testing proxy endpoint information... 2024-04-24 20:37:49.519915+0200 client[57750:9051218] dynamic_cast error 2: One or more of the following type_info's has hidden visibility or is defined in more than one translation unit. They should all have public visibility. N3Ice12EndpointInfoE, N11IceInternal5InfoIIN3Ice15TCPEndpointInfoEEE, N3Ice15TCPEndpointInfoE.
2024-04-24 20:37:49.520080+0200 client[57750:9051218] dynamic_cast error 2: One or more of the following type_info's has hidden visibility or is defined in more than one translation unit. They should all have public visibility. N3Ice12EndpointInfoE, N11IceInternal5InfoIIN3Ice15TCPEndpointInfoEEE, N3Ice15TCPEndpointInfoE.
2024-04-24 20:37:49.520088+0200 client[57750:9051218] dynamic_cast error 2: One or more of the following type_info's has hidden visibility or is defined in more than one translation unit. They should all have public visibility. N3Ice12EndpointInfoE, N11IceInternal5InfoIIN3Ice15UDPEndpointInfoEEE, N3Ice15UDPEndpointInfoE.
```

### PR DESCRIPTION
**Pull request by bernardnormier**
This PR adds a public/exported out of line dtor to all EndpointInfo and ConnectionInfo.

Fixes #2071.

### CODE REVIEW

File: cpp/src/Ice/Connection.cpp [hunk 1]
```diff
@@ -7,6 +7,12 @@
 using namespace Ice;
 using namespace std;
 
+Ice::ConnectionInfo::~ConnectionInfo() {}
```
Comment (externl) [inline cpp/src/Ice/Connection.cpp:10]: Should we have a comment `// out of line to avoid weak vtable`

File: cpp/src/IceBT/EndpointI.cpp [hunk 2]
```diff
@@ -20,6 +20,9 @@ using namespace std;
 using namespace Ice;
 using namespace IceBT;
 
+IceBT::ConnectionInfo::~ConnectionInfo() {}
```
Comment (externl) [inline cpp/src/IceBT/EndpointI.cpp:23]: Ditto for the rest

**Review by externl — APPROVED** [review]

File: cpp/src/IceIAP/EndpointI.mm [hunk 3]
```diff
@@ -77,6 +77,9 @@ ICEIAP_API void registerIceIAP(bool loadOnInitialize)
     }
 }
 
+IceIAP::ConnectionInfo::~ConnectionInfo() {}
```
Comment (pepone) [inline cpp/src/IceIAP/EndpointI.mm:80]: We need to include ConnectionInfo.h

**Review by pepone — APPROVED** [review]

File: cpp/src/IceIAP/EndpointI.mm (hunk 3 above)
Comment (bernardnormier) [reply cpp/src/IceIAP/EndpointI.mm:82]: Fixed

**Pull request merged**